### PR TITLE
Fix problem reading AASTex table without trailing backslashes at end

### PR DESCRIPTION
**Pull request by taldcroft**
Closes #5236.  (By superceding).

### CODE REVIEW

**Comment by taldcroft** [comment]
cc: @hamogu

File: astropy/io/ascii/latex.py [hunk 1]
```diff
@@ -80,13 +82,21 @@ class LatexSplitter(core.BaseSplitter):
...
+            print(last_line)
+            lines[-1] = last_line + r'\\'
+
+        return super(LatexSplitter, self).__call__(lines)
+
     def process_line(self, line):
         """Remove whitespace at the beginning or end of line. Also remove
         \\ at end of line"""
-        line = line.split('%')[0]
+        line = RE_COMMENT.split(line)[0]
```
Comment (hamogu) [inline astropy/io/ascii/latex.py]: I don't know enough about regexpes: What is the line has white space list so: `data1 & data2 \\     %and now a comment``?

File: astropy/io/ascii/latex.py [hunk 2]
```diff
@@ -80,13 +82,21 @@ class LatexSplitter(core.BaseSplitter):
...
         line = line.strip()
-        if line[-2:] == r'\\':
-            line = line.strip(r'\\')
+        if line.endswith(r'\\'):
+            line = line.rstrip(r'\\')
```
Comment (taldcroft) [inline astropy/io/ascii/latex.py]: @MSeifert04 - yup, that's correct.  But in the interest not clogging up CI even more I'm merging this as-is.  After the 1.3 frenzy quiets we can do this clean up.

**Comment by eteq** [comment]
backported to v1.0.x in 1f7cb648c5529f75917878783ec9edf6fa1dad86

**Pull request merged**